h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstRowSlice.java

```diff
@@ -98,7 +98,9 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
         }
       }.doAll(fr.types(), fr).outputFrame(fr.names(), fr.domains());
     } else if ((asts[2] instanceof AstNum)) {
-      long[] rows = new long[]{(long) (((AstNum) asts[2]).getNum())};
+      double num = ((AstNum) asts[2]).getNum();
+      if(num < 0) throw new IllegalStateException("Checking whether deepSlice with negative indices is being used "); 
```

Comment:
so the purpose is to temporarily raise an exception until this is fixed, am I correct?
Can you please just improve the message and use sth like "Unsupported negative index in deepSlice"?

Reply:
@seb-h2o right. I can improve msg, np. Could you please confirm that it is bug. If we agree that it is a bug then we can proceed and fix it.

Reply:
@seb-h2o it is just execution will not reach `fail` statement anyway due to an error before. This `fail` is just a comment.